Samples/Samples/View/CalendarPage.xaml

```diff
@@ -1,49 +1,172 @@
-                    Text="Load Calendars" 
-                    Command="{Binding GetCalendars}"/>
-        </Grid>
-        <ListView Grid.Row="1" 
-                  ItemsSource="{Binding Calendars}"
-                  ItemTemplate="{StaticResource CalendarItemTemplate}"/>
+              </Grid.ColumnDefinitions>
+              <Label Text="{Binding Title}"
+                     TextColor="Black"
+                     HorizontalTextAlignment="Center"/>
+              <Label Grid.Column="1"
+                     Text="{Binding Start, Converter={StaticResource EpochToDateTimeConverter}}"
+                     TextColor="Black"
+                     HorizontalTextAlignment="Center"/>
+              <Label Grid.Column="2"
+                     Text="-"/>
+              <Label Grid.Column="3"
+                     Text="{Binding End, Converter={StaticResource EpochToDateTimeConverter}}"
+                     TextColor="Black"
+                     HorizontalTextAlignment="Center"/>
+            </Grid>
+          </StackLayout>
+        </ViewCell>
+      </DataTemplate>
+    </ResourceDictionary>
+  </views:BasePage.Resources>
+  <views:BasePage.BindingContext>
+    <viewmodels:CalendarViewModel />
+  </views:BasePage.BindingContext>
+  <views:BasePage.Content>
+    <Grid>
+      <Grid.RowDefinitions>
+        <RowDefinition Height="Auto"/>
+        <RowDefinition Height="Auto"/>
+        <RowDefinition Height="Auto"/>
+        <RowDefinition Height="Auto"/>
+        <RowDefinition/>
+      </Grid.RowDefinitions>
+      <Grid>
+        <Grid.ColumnDefinitions>
+          <ColumnDefinition/>
+          <ColumnDefinition/>
+          <ColumnDefinition/>
+        </Grid.ColumnDefinitions>
+        <Button Text="Request Read Access"
+                Command="{Binding RequestCalendarReadAccess}"
+                HeightRequest="100"/>
+        <Button Grid.Row="0"
+                Grid.Column="1"
+                Text="Request Write Access"
+                Command="{Binding RequestCalendarWriteAccess}"/>
+        <Button Grid.Row="0"
+                Grid.Column="2"
+                Text="Load Calendars"
+                Command="{Binding GetCalendars}"/>
+      </Grid>
+      <Picker Grid.Row="1"
+              Title="Select a Calendar"
+              ItemsSource="{Binding Calendars}"
+              ItemDisplayBinding="{Binding Name}"
+              SelectedItem="{Binding SelectedCalendar, Mode=TwoWay}"/>
+      <Grid Grid.Row="2">
+        <Grid.RowDefinitions>
+          <RowDefinition Height="Auto"/>
+          <RowDefinition Height="Auto"/>
+        </Grid.RowDefinitions>
+        <Grid.ColumnDefinitions>
+          <ColumnDefinition Width="Auto"/>
+          <ColumnDefinition/>
+          <ColumnDefinition/>
+          <ColumnDefinition/>
+        </Grid.ColumnDefinitions>
+        <CheckBox Grid.Row="0"
+                  Grid.Column="0"
+                  IsChecked="{Binding StartDatePickersEnabled}">
```

Comment:
I'd change this to be 2-way binding and use that to trigger the update. Alternatively, you could introduce a load button and only load calendar items when they click the button.